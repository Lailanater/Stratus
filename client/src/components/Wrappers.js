import React from 'react';
import { useSelector } from 'react-redux';
import { ThemeProvider } from '@material-ui/styles';
import { SnackbarProvider } from 'notistack';
import createMuiTheme from '@material-ui/core/styles/createMuiTheme';
import { Auth0Provider } from '../react-auth0-spa';

const Wrappers = props => {
  const theme = useSelector(state => state.theme);
  let currentTheme = createMuiTheme({
    palette: {
      type: theme,
      primary: { main: '#00838F' },
      secondary: { main: '#ed407a' }
    }
  });

  return (
    <ThemeProvider theme={currentTheme}>
      <Auth0Provider>
        <SnackbarProvider maxSnack={2}>{props.children}</SnackbarProvider>
      </Auth0Provider>
    </ThemeProvider>
  );
};

export default Wrappers;
