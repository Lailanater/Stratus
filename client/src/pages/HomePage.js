import React from 'react';
import ButtonMenu from "../components/ButtonMenu";
import CurrentProjectDropdown from "../components/CurrentProjectDropdown";

const HomePage = (props) => {
    return (
        <div className="container">
            <CurrentProjectDropdown />
            <br />
            <ButtonMenu />
        </div>
    );
};

export default HomePage;