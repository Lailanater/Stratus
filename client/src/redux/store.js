import { createStore } from 'redux';
import rootReducer from './reducers';
import { loadState } from '../localStorage';

let initialState = loadState();

if (initialState === undefined) {
  initialState = {
    isAppMenuDisplayed: false,
    theme: 'dark',
    currentProject: '',
    projects: [],
    menuName: '',
    needsRepeat: false,
    needsDtmf: false,
    needsVoice: false,
    dtmfOptions: [],
    defaultRouteTo: 'menu',
    isNextButtonDisabled: false,
    isFinishButtonDisabled: false
  };
}

export default createStore(rootReducer, initialState);
