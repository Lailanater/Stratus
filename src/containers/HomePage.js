import React from 'react';
import ButtonMenu from "../components/ButtonMenu";

function HomePage() {
    return (
        <div>
            <ButtonMenu/>
        </div>
    );
}

export default HomePage;