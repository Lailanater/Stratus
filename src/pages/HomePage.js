import React from 'react';
import ButtonMenu from "../components/ButtonMenu";
import CurrentProjectDropdown from "../components/CurrentProjectDropdown";

const HomePage = (props) => {
    return (
        <div>
            <CurrentProjectDropdown/>
            <ButtonMenu/>
        </div>
    );
};

export default HomePage;