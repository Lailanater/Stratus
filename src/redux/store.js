import {createStore} from "redux";
import rootReducer from "./reducers";

export const initialState = {
    isAppMenuDisplayed: false,
    theme: "dark",
    currentProject: "",
    projects: []
};

export default createStore(rootReducer);