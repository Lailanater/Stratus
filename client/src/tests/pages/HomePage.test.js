import React from 'react';
import renderer from 'react-test-renderer';
import HomePage from '../../pages/HomePage';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import Wrappers from '../../components/Wrappers';

it('renders without crashing', () => {
  const component = renderer.create(
    <Provider store={store}>
      <Wrappers>
        <HomePage />
      </Wrappers>
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
