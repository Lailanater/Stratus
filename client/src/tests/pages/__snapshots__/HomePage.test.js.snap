// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="container"
>
  <div
    className="MuiPaper-root MuiPaper-elevation5 container MuiPaper-rounded"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated"
    >
      Current Project
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline"
      onClick={[Function]}
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby=" "
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </div>
      <input
        onAnimationStart={[Function]}
        type="hidden"
      />
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
  <br />
  <div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "#192a56",
          "color": "white",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Create Menu
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "#40739e",
          "color": "white",
          "marginLeft": "5px",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Create Grammar
      </span>
    </button>
  </div>
</div>
`;
